### PR TITLE
Use zig as c compiler for cross platform builds

### PR DESCRIPTION
**Pull request by jfantinhardesty**
### What type of Pull Request is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

### Describe your changes in brief

Changes our C compiler to use Zig https://ziglang.org/, which allows easier cross compilation support. This should allow us to deploy Cloudfuse to more Linux OS's without needing to change our deployment strategy or spin up VM's and docker images.

### Checklist

- [x] Tested locally
- [ ] Added new dependencies
- [ ] Updated documentation
- [ ] Added tests

### Related Issues

- Related Issue #
- Closes #

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -192,8 +193,15 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      -
-        name: Install Go
+        
+      - name: Install Zig
```
Comment (foodprocessor) [inline .github/workflows/publish-release.yml:197]: Did you do a test run of the release in GitHub Actions to verify it all works?
I think the way I did this last time I tried was by changing the trigger to include pushes to this branch. And I told GoReleaser to skip version verification and publishing.

**Review by foodprocessor — APPROVED** [review]
🎉

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (jfantinhardesty) [reply .github/workflows/publish-release.yml:199]: Yes I did do a test run, which caught a few issues, but now it works.

**Review by foodprocessor — APPROVED** [review]
Double approved!

**Pull request merged**